File: Blood-Donor/Blood-Donor-FrontEnd/src/services/donorService.js
import { apiRequest } from './apiClient';

export async function searchDonors(bloodGroup, pinCode, limit = 50, nextCursor = null) {
  const params = new URLSearchParams({
    bloodGroup,
    pinCode,
    limit: String(limit),
  });

  if (nextCursor) {
    params.set('nextCursor', nextCursor);
  }

  const res = await apiRequest(`/donors/search?${params.toString()}`);
  return res.data;
}

export async function searchDonorsNearby({
  bloodGroup,
  latitude,
  longitude,
  radiusKm = 25,
  pinCode = '',
  limit = 50,
}) {
  const params = new URLSearchParams({
    bloodGroup,
    latitude: String(latitude),
    longitude: String(longitude),
    radiusKm: String(radiusKm),
    limit: String(limit),
  });

  if (pinCode) {
    params.set('pinCode', pinCode);
  }

  const res = await apiRequest(`/donors/search?${params.toString()}`);
  return res.data;
}

File: Blood-Donor/Blood-Donor-Backend/src/main/java/com/blooddonor/service/DonorService.java
package com.blooddonor.service;

import com.blooddonor.dto.request.DonorUpdateRequest;
import com.blooddonor.dto.response.CursorDonorSearchResponse;
import com.blooddonor.dto.request.LiveLocationRequest;
import com.blooddonor.dto.response.DonorDashboardResponse;
import com.blooddonor.dto.response.DonorResponse;

public interface DonorService {

    DonorResponse getProfile();

    DonorDashboardResponse getDashboard();

    DonorResponse updateProfile(DonorUpdateRequest request);

    DonorResponse updateLiveLocation(LiveLocationRequest request);

    void deleteAccount();

    CursorDonorSearchResponse searchDonors(
            String bloodGroup,
            String pinCode,
            Double latitude,
            Double longitude,
            Double radiusKm,
            int limit,
            String nextCursor,
            String previousCursor);
}
